src/main/scala/de/frosner/ddq/core/Check.scala

```diff
@@ -563,6 +573,41 @@ object Check {
   )
 
   /**
+   * Check whether columns in the dependent set have functional dependency on determinant set.
+   * @param determinantSet sequence of column names which form a determinant set
+   * @param dependentSet sequence of column names which form a dependent set
+   * @return [[core.Constraint]] object
+   */
+  def hasFunctionalDependency(determinantSet: Seq[String], dependentSet: Seq[String]) = Constraint(
+    df => {
+      val determinantIndexes = df.columns.zipWithIndex.filter {
```

Comment:
After I removed this filter most of the tests are failing because you always used column references that do not exist (like `column0`). What was your intention behind that?